packages/rum/src/domain/record/observers/mutationObserver.spec.ts

```diff
@@ -883,16 +847,16 @@ describe('startMutationCollection', () => {
 
         const { mutationCallbackSpy } = startMutationCollection()
 
-        hiddenElement.appendChild(document.createTextNode('function foo() {}'))
+        append('function foo() {}', hiddenElement)
 
         flushMutations()
 
         expect(mutationCallbackSpy).not.toHaveBeenCalled()
       })
 
       it('when mutating a known child node', () => {
-        const textNode = document.createTextNode('function foo() {}')
-        sandbox.appendChild(textNode)
+        const textNode = append<Text>('function foo() {}', sandbox)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏it might have been better to keep `appendText` / `appendElement` so we don't need to "cast" the return like this. We could also imagine using TS to infer this properly: https://www.typescriptlang.org/play?#code/FAFwngDgpgBA8gVxBJAeAKjKAPEUB2AJgM4zEgBOAlvgOYB8MAvMDG5jnkaQAYwAkAbxoAzKBRgUo5AL58A-PCQoQqKeUYAuVuyy4CJGDwA6+IaPGTpIOTEWJkadSC06YHfdyOpz+MRIB9W0UAUQAbKABbAhAYbTZ3PS5DHl9-GCCAbRAAQwpaKBAAXTTLIIUYcKiYuLd2TmBgQigAYzC82BEEfBaQKgB7fBgciGgiDCSDUnJqOnoACgALEEiwzXcASnWHFQx6AG5GlsHyLAjo-FimYdGDeYAiVBB+5-p7jeAAek+EtgA9eSNYDHfCnKDnGIAJmYNzGhAeCSeL36bw+31+MABQJBpzwuBhIzhCLYz2e7y+P1+WKOJ1i4OqlwAzATbkQHqT+jAkc8YNk8gViqiKRisUA

Reply:
Okay, why not keeping separating `appendText` and `appendElement` however there are also other cases where we need to cast to specify which HTMLElement we want and I don't know if a smart inferring worth it. How would you handle them? 
cf: https://github.com/DataDog/browser-sdk/blob/8dc1b90601b8bf398d6604ead3cbeb4534f92937/packages/rum/src/domain/record/record.spec.ts#L298

Reply:
I would just cast them `as HTMLDivElement`.

Functions like `function foo<T>(): T` are implicitly casting the return value anyway.

Reply:
I splitted to have appendText and appendElement